app/src/main/java/com/framgia/fdms/screen/producer/vendor/VendorViewModel.java

```diff
@@ -116,14 +136,45 @@ public void onLoadVendorFailed() {
 
     @Override
     public void onEditSubmitClick(Producer vendor) {
+        if (TextUtils.isEmpty(vendor.getName())) {
+            setMessageError(mActivity.getString(R.string.msg_error_user_name));
+            return;
+        }
         mVendorDialog.dismiss();
+        switch (mTypeAction) {
+            case ACTION_EDIT_VENDOR:
+                mVendorEdit.setName(vendor.getName());
+                mVendorEdit.setDescription(vendor.getDescription());
+                break;
+            case ACTION_ADD_VENDOR:
+                mVendors.add(0, vendor);
+                mAdapter.notifyItemInserted(0);
+                setPositionScroll(0);
+                break;
+            default:
+                break;
+        }
     }
 
     @Override
     public void onEditCancelClick(Producer vendor) {
         mVendorDialog.dismiss();
     }
 
+    @Override
+    public void onAddVendorClick() {
+        mTypeAction = ACTION_ADD_VENDOR;
+        mVendorDialog = ProducerDialog.newInstant(this, new Producer());
+        mVendorDialog.show(mActivity.getSupportFragmentManager(),
+            Constant.BundleConstant.BUNDLE_DEVICE);
+    }
+
+    @IntDef({ACTION_EDIT_VENDOR, ACTION_ADD_VENDOR})
+    public @interface Action {
+        int ACTION_EDIT_VENDOR = 0;
```

Comment:
ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.